daml/Marketplace/Custody/Model.daml

```diff
@@ -120,23 +121,41 @@ template LifecycleRequest
           when (postAssetVersion /= assetDeposit.asset.id.version) $
             createNewVersionOfDescription description postAssetVersion serializedRemaining
           let payoutSplitQuantities = (\(qty, _) -> qty * assetDeposit.asset.quantity) <$> pending
-          -- Next step could be:
-          -- create SettlementInstruction with buyer = customer; seller = issuer; ..
           splitResultDeposits <- exercise safekeepingDepositCid AssetDeposit_Split with quantities = payoutSplitQuantities
           let payoutCids = init splitResultDeposits
-          payoutCids <- forA payoutCids \payoutDepositCid -> do
-            exerciseByKey @AssetSettlementRule assetDeposit.account.id $
-               AssetSettlement_AddController description.issuer
-            cid <- exerciseByKey @AssetSettlementRule description.safekeepingAccountId (
-              AssetSettlement_Transfer with
-                receiverAccountId = assetDeposit.account.id
-                depositCid = payoutDepositCid)
-            exerciseByKey @AssetSettlementRule assetDeposit.account.id $
-              AssetSettlement_RemoveController description.issuer
-            pure cid
+              masterAgreementSignatories = [provider, customer, investor]
+          payoutInstructions <- createPayoutInstructions payoutCids description assetDeposit operator masterAgreementSignatories
           assetDepositCid' <- create assetDeposit with asset = (assetDeposit.asset with id = (assetDeposit.asset.id with version = postAssetVersion))
           archive assetDepositCid
-          return $ assetDepositCid' :: payoutCids
+          return (assetDepositCid', payoutInstructions)
+
+createPayoutInstructions : [ContractId AssetDeposit] -> AssetDescription -> AssetDeposit -> Party -> [Party] -> Update [ContractId SettlementInstruction]
```

Comment:
I would avoid creating top-level fns that are not meant to be part of the module API; just inline them using `let` inside `Process`. They're only used once, right? Maybe even inline the whole `createPayoutInstructions`, it may make it more legible.

Reply:
In my opinion putting such stuff into let leads to long, unreadable "messy" code, and let on choice level (not template level) even more. Of course this is for function bodies.

They are only used once, the reasoning behind adding such functions was that I wanted to make it readable/cleaner/more compact. It was already a bit long, I wanted to either reduce that (we can say boy scouting) or at least not increase. :)

As you are the main code owner of this, I can accept if you strongly insist. :)

Reply:
Also note that most possibly one is only interested at the level of "Process" that there are a couple of payout settlement instructions created. I wanted to reflect this and hide the details which are only important at a lower abstraction level - in this function.
This is true for the other functions as well, e.g. createTradeId could be tested on its own (that it creates a unique Id gives this or that).

Reply:
That's fair enough. If you keep them at the top-level, please do update `module x where (...)` to make clear these are not part of the public API.

Reply:
Okay good idea.
My team mates pointed out that unfortunately hiding any function with `module x where (...)` still leaves it accessible in lf (e.g. you import the dar and you can use anything unexported).

Reply:
Done.